dash-spv/src/storage/disk/state.rs

```
@@ -424,9 +425,9 @@ impl DiskStorageManager {
     }
 
     /// Shutdown the storage manager.
-    pub async fn shutdown(&mut self) -> StorageResult<()> {
-        // Save all dirty segments
-        super::segments::save_dirty_segments(self).await?;
+    pub async fn shutdown(&mut self) {
```

Shouldn't we keep the returned result and instead handle the errors we discard currently.

---

The only thing returning an error here previously was the save dirty segments but it no longer does. If we want to return an error when send or when join handle fails that would be other PR addressing the background worker related logic. Also, I am not sure what is the best way to proceed if something like that happens. Maybe it only return error if the worker was aborted, in that case it would be right to consider the shutdown call a success